src/AutoRest.CSharp/Common/Input/InputTypes/InputDateTimeType.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace AutoRest.CSharp.Common.Input;
+
+internal record InputDateTimeType(DateTimeKnownEncoding Encode, InputPrimitiveType WireType, bool IsNullable) : InputType("DateTime", IsNullable)
+{
+    public static InputDateTimeType Rfc3339 { get; } = new(DateTimeKnownEncoding.Rfc3339, InputPrimitiveType.String, false);
```

Comment:
Same here, shall we allow the user to define nullability here?

Reply:
answered in the other comment: https://github.com/Azure/autorest.csharp/pull/4707#discussion_r1621603410